srv/salt/_modules/osd.py

```diff
@@ -573,30 +573,31 @@ def wait(self):
         """
         i = 0
         last_pgs = 0
+        fmt_msg = msg = "osd.{} has {} PGs remaining"
         while i < self.settings['timeout']/self.settings['delay']:
-            rc, msg = self.osd_safe_to_destroy()
+            rc, out = self.osd_safe_to_destroy()
             if rc == 0:
-                ret = "osd.{} is safe to destroy".format(self.osd_id)
-                log.info(ret)
-                return ret
+                msg = "osd.{} is safe to destroy".format(self.osd_id)
+                log.info(msg)
+                return msg
+
             entry = self.osd_df()
             if 'pgs' in entry:
+                msg = fmt_msg.format(self.osd_id, entry['pgs'])
```

Comment:
I generally prefer using the dict's `.get` method to properly fallback to a sane default and avoid IndexErrors. 

That whole `pgs in entry` conditional can be simplified using the mentioned notation.

Reply:
Agreed, branch has been force-pushed with this change.